gms/client/src/main/java/com/linkedin/entity/client/EntityClient.java

```diff
@@ -24,240 +19,232 @@
+        .startParam(start)
+        .limitParam(limit);
+    if (requestFilters != null) {
+      requestBuilder.filterParam(newFilter(requestFilters));
     }
+    return sendClientRequest(requestBuilder.build()).getEntity();
+  }
+
+  public void update(@Nonnull final Entity entity) throws RemoteInvocationException {
+    // TODO: Replace with EntitiesDoIngestActionBuilder.
+
+    final FieldDef<?> entityFieldDef =
+        new FieldDef<>(PARAM_ENTITY, Entity.class, DataTemplateUtil.getSchema(String.class));
+
+    final HashMap<String, DynamicRecordMetadata> actionRequestMetadata = new HashMap<>();
+    actionRequestMetadata.put(ACTION_INGEST, new DynamicRecordMetadata(ACTION_INGEST, Arrays.asList(entityFieldDef)));
+
+    final HashMap<java.lang.String, DynamicRecordMetadata> actionResponseMetadata = new HashMap<>();
+    actionResponseMetadata.put(ACTION_INGEST, new DynamicRecordMetadata(ACTION_INGEST, Collections.emptyList()));
+
+    final ResourceSpec resourceSpec =
+        new ResourceSpecImpl(Collections.emptySet(), actionRequestMetadata, actionResponseMetadata, String.class,
+            EmptyRecord.class, EmptyRecord.class, EmptyRecord.class, Collections.emptyMap());
+
+    final ActionRequestBuilder builder =
+        new ActionRequestBuilder(RESOURCE_NAME, Void.class, resourceSpec, RestliRequestOptions.DEFAULT_OPTIONS);
+
+    builder.name("ingest");
+    builder.addParam(entityFieldDef, entity);
+
+    final Request request = builder.build();
+
+    sendClientRequest(request);
+  }
+
+  /**
+   * Searches for datasets matching to a given query and filters
+   *
+   * @param input search query
+   * @param requestFilters search filters
+   * @param start start offset for search results
+   * @param count max number of search results requested
+   * @return Snapshot key
+   * @throws RemoteInvocationException
+   */
+  @Nonnull
+  public SearchResult search(@Nonnull String entity, @Nonnull String input, @Nonnull Map<String, String> requestFilters,
+      int start, int count) throws RemoteInvocationException {
+
+    return search(entity, input, null, requestFilters, start, count);
+  }
+
+  @Nonnull
+  public SearchResult search(@Nonnull String entity, @Nonnull String input, @Nullable StringArray aspectNames,
+      @Nullable Map<String, String> requestFilters, int start, int count) throws RemoteInvocationException {
+
+    final EntitiesDoSearchRequestBuilder requestBuilder = ENTITIES_REQUEST_BUILDERS.actionSearch()
+        .entityParam(entity)
+        .inputParam(input)
+        .filterParam(newFilter(requestFilters))
+        .startParam(start)
+        .countParam(count);
+
+    return sendClientRequest(requestBuilder.build()).getEntity();
+  }
+
+  /**
+   * Gets browse path(s) given dataset urn
+   *
+   * @param urn urn for the entity
+   * @return list of paths given urn
+   * @throws RemoteInvocationException
+   */
+  @Nonnull
+  public StringArray getBrowsePaths(@Nonnull Urn urn) throws RemoteInvocationException {
```

Comment:
This appears to be just formatting -- is that correct?

Reply:
Yeah :( Ran formatter for my code. and it ran for everything.

Reply:
The only logic is to take out the key aspect generation logic into the utils